src/mimefactory.rs

```diff
@@ -562,11 +594,18 @@ impl<'a> MimeFactory<'a> {
         };
 
         // Store protected headers in the inner message.
-        let mut message = protected_headers
+        let message = headers
+            .protected
             .into_iter()
             .fold(message, |message, header| message.header(header));
```

Comment:
as pointed out by @link2xt at https://github.com/deltachat/deltachat-core-rust/pull/2232#issuecomment-778701339 , these protected headers are needed in the outermost imf-header for unencrypted messages.

however, the logic below (last `else` branch) will move them to the first body-part-header, where they are not readable by current delta chats - and also not by other mua, which results in messages without a subject (the subject is an protected header, however, must be moved there only when encryption takes place).

probably only a tiny change is needed to correct that, however, with a large impact then :)

Reply:
I pushed a fixup after your test.